HW_23/mongo_collection.py

```diff
@@ -0,0 +1,21 @@
+from client import BaseDB
+
+
+class MongoCollection(BaseDB):
+    def __init__(self, collection):
+        super().__init__(collection)
+
+    def insert_one(self, dictionary):
+        return self.collection.insert_one(dictionary)
+
+    def insert_many(self, list_of_dicts):
+        return self.collection.insert_many(list_of_dicts)
+
+    def find_one(self, data):
+        return self.collection.find_one(data)
+
+    def find_all(self):
+        return self.collection.find()
+
+    def find_by_query(self, query):
+        return self.collection.find(query)

```

Comment:
what is the difference between this class and BaseDb?

Reply:
There is a slight differense: in BaseDB class I defined client and db data, in MongoCollection class I defined a collection. 
By the way, I don't understand, why we should have 2 classes with inheritance for this, if it can be implemented with one only.

Reply:
But all described functions here are the same as in BaseDB so you do not need it. You can use them from the Base class.  In BaseClass we describe general calls, we can add some logging there. And than we can implement classes for different collections but with more readable names and some additional logic using BaseClass. 

For Example:
We have Base as you described. And have collection for employees. 
So we implement:
class EmploeeCollection(Base):
         def get_employee(self, employee_id):
                return self.find_one({'_id': employee_id})

         # and in some cases we need only specific data from this employee so we can implement
         def get_employee_bonus_and_manager(self, id):
               employee = self.find_one({'_id': employee_id})
                return employee['bonus'], employee['manager']


And so on. But it is only one way of implementation. It can be different from project to project depends on what you need.

Reply:
Many thanks for an explanation, now I understand